src/include/stir/recon_buildblock/PLSPrior.h

```diff
@@ -0,0 +1,214 @@
+  These are the keywords that can be used in addition to the ones in GeneralPrior.
+  \verbatim
+  PLS Prior Parameters:=
+  ; next defaults to 0, set to 1 for 2D images, 0 for 3D
+  only 2D:= 0
+
+
+  anatomical_filename:= file.hv; Image that provides anatomical information (i.e. CT or MR image). The
+                   dimension should be the same as that of the emission image.
+
+  eta :=    ; A parameter for preventing the division by zero problem. The value dependes
+                   on the scale of the anatomical image.
+
+  alpha :=   ; A parameter that controls the edge-preservation property of PLS. The value
+                   depends on the scale of the emission image.
+
+  ; use next parameter to specify an image with penalisation factors (a la Fessler)
+  ; see class documentation for more info
+  ; kappa filename:=
+  ; use next parameter to get gradient images at every subiteration
+  ; see class documentation
+  gradient filename prefix:=
+  END PLS Prior Parameters:=
+  \endverbatim
+
+
+*/
+template <typename elemT>
+class PLSPrior:  public
+                       RegisteredParsingObject< PLSPrior<elemT>,
+                                                GeneralisedPrior<DiscretisedDensity<3,elemT> >,
+                                                GeneralisedPrior<DiscretisedDensity<3,elemT> >
+                                              >
+{
+ private:
+  typedef
+    RegisteredParsingObject< PLSPrior<elemT>,
+                             GeneralisedPrior<DiscretisedDensity<3,elemT> >,
+                             GeneralisedPrior<DiscretisedDensity<3,elemT> > >
+    base_type;
+
+ public:
+  //! Name which will be used when parsing a GeneralisedPrior object
+  static const char * const registered_name;
+
+  //! Default constructor
+  PLSPrior();
+
+  //! Constructs it explicitly
+  PLSPrior(const bool only_2D, float penalization_factor);
+
+
+  //! compute the value of the function
+  double
+    compute_value(const DiscretisedDensity<3,elemT> &current_image_estimate);
+
+  //! compute gradient
+  void compute_gradient(DiscretisedDensity<3,elemT>& prior_gradient,
+                        const DiscretisedDensity<3,elemT> &current_image_estimate);
+
+  //! get penalty weights for the neigbourhood
+  Array<3,float> get_weights() const;
+
+  //! set penalty weights for the neigbourhood
+  void set_weights(const Array<3,float>&);
+
+  //! get current kappa image
+  /*! \warning As this function returns a shared_ptr, this is dangerous. You should not
+      modify the image by manipulating the image refered to by this pointer.
+      Unpredictable results will occur.
+  */
+  shared_ptr<DiscretisedDensity<3,elemT> > get_kappa_sptr() const;
+  shared_ptr<DiscretisedDensity<3,elemT> > get_anatomical_grad_sptr(int direction) const;
+  shared_ptr<DiscretisedDensity<3,elemT> > get_norm_sptr() const;
+
+  //! set kappa image
+  void set_kappa_sptr(const shared_ptr<DiscretisedDensity<3,elemT> >&);
+
+  //! set anatomical pointers
```

Comment:
I find these rather dangerous as public members. it'd be hard to make it consistent. Instead it would be good/essential to have a `set_anatomical_image_sptr` (and potentially a `get`)

Similarly we'll need `get/set_eta/alpha` for putting this into SIRF.